### ISSUE TITLE
Losing headers when using Undici in latest Node 18.x

### ISSUE THREAD

**Issue by joshkel** [mswjs/headers-polyfill#72]
Node.js's update to Undici 5.26.3 caused its `Headers` class to be called `_Headers`:

https://github.com/nodejs/node/pull/50153/files#diff-f516ab824a7722da938a4c7c851520d39731ddeb4f7198dff4e932c5d4f8fdf7

As a result, passing a native Undici Headers object to headers-polyfill (which happens when MSW intercepts a native Node.js `fetch` request with headers) loses the headers' contents.

To reproduce, run the following script in Node.js 18.16.1 (where it outputs the expected headers) and Node 18.18.2 (where it does not).

```js
const hp = require('headers-polyfill');

console.log(hp.headersToString(new hp.Headers(new Headers({'Content-Type': 'application/json'}))));
```

**Comment by kettanaito** [comment]
## Released: v4.0.2 🎉

This has been released in v4.0.2!

- 📄 [**Release notes**](https://github.com/mswjs/headers-polyfill/releases/tag/v4.0.2)
- 📦 [npm package](https://www.npmjs.com/package/headers-polyfill/v/4.0.2)

Make sure to always update to the latest version (`npm i headers-polyfill@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Comment by themagickoala** [comment]
We are still using MSW v1, and we have applied the override here to make this work. However, there is another issue: _Headers does not have an `all` method in this version of headers-polyfill (as of 3.3.0 actually). This is called to get headers in SetupServerApi.init, and this is throwing an error. It seems that MSW v1 is still relying on this older headers-polyfilll, and as such is not supported for Node v18.18.2+, which requires upgrading to headers-polyfill v4.0.2?

Would it be possible to put out a new release for MSW v1, or will we have to upgrade to v2 to resolve this?
